### PR TITLE
Pip 1358 use service account for compute

### PR DESCRIPTION
**Pull request by leepc12**
See https://encodedcc.atlassian.net/browse/PIP-1358 for details.

This PR was originally created to fix service account issue (not using given service account `--gcp-service-account-key-json` for provisioning instances on Google Compute Engine). It was simply fixed by adding a few lines to the backend file (`backend.conf`).
https://github.com/ENCODE-DCC/caper/pull/88/files#diff-5bbfdab5434768780bed99ad70362643R309

However, found a critical hanging issue (related to early closing of file object for logging) so fix for such issue is also included in this PR.

Also improved warning for user interruption (Ctrl + C).

### CODE REVIEW

**Review by ottojolanki — CHANGES_REQUESTED** [review]
Looks good, some naming changes requested. I really like the exception reraising decorator!

File: caper/caper_workflow_opts.py [hunk 1]
```diff
@@ -183,13 +183,18 @@ def create_file(
...
                 logger.info(
-                    'Docker image found in WDL. wdl={wdl}, d={d}'.format(
+                    'Docker image found in WDL\'s metadata. wdl={wdl}, d={d}'.format(
                         wdl=wdl, d=docker
                     )
                 )
             else:
                 logger.warning(
-                    'Docker image not found in WDL. wdl={wdl}'.format(wdl=wdl)
+                    'Docker image not found in WDL\'s metadata, which means that '
```
[the remark this answers is not in the document]
Comment (leepc12) [reply caper/caper_workflow_opts.py]: Done as you suggested. I am actually planning to replace all single-quotes with double-quotes in the next refactoring of the code.

File: caper/cromwell_rest_api.py [hunk 2]
```diff
@@ -9,6 +9,30 @@
 logger = logging.getLogger(__name__)
 
 
+def request_error_handler(func):
```
[the remark this answers is not in the document]
Comment (leepc12) [reply caper/cromwell_rest_api.py]: Done as you suggested.

File: caper/cli.py [hunk 3]
```diff
@@ -306,30 +311,33 @@ def subcmd_server(caper_runner, args, nonblocking=False):
             heartbeat_timeout=args.server_heartbeat_timeout,
         )
 
+    kwargs = {
```
[the remark this answers is not in the document]
Comment (leepc12) [reply caper/cli.py]: Fixed. Changed it to `args_from_cli`.

**Review by ottojolanki — APPROVED** [review]
LGTM

**Pull request merged**